jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js

```diff
@@ -250,15 +261,19 @@ YUI.add('deployment-flow', function() {
       @method _handleDeploy
     */
     _handleDeploy: function() {
-      const credential = this.state.credential;
-      const cloud = this.state.cloud && this.state.cloud.name || null;
-      const region = this.state.region;
       let modelName = '';
       if (this.refs.modelName) {
         modelName = this.refs.modelName.getValue();
       }
-      this.props.deploy(
-        this._handleClose, true, modelName, credential, cloud, region);
+      const args = {
+        credential: this.state.credential,
+        cloud: this.state.cloud && this.state.cloud.name || undefined,
+        region: this.state.region
+      };
+      if (this.state.sshKey) {
+        args.config = {'authorized-keys': this.state.sshKey};
+      }
+      this.props.deploy(this._handleClose, true, modelName, args);
```

Comment:
the api for creating a new model is deploy? Is that Juju's API for it or our own?